### ISSUE TITLE
Behavior broken after resizing window when using "Auto Slides Per View / Carousel Mode"

### ISSUE THREAD

**Issue by italosimpli** [nolimits4web/swiper#3674]
I believe this is related to [this comment](https://github.com/nolimits4web/swiper/issues/2095#issuecomment-360385564) in the issue [#2095](https://github.com/nolimits4web/swiper/issues/2095).

This is a (multiple allowed):
* [x] bug
* [x] enhancement
* [x] feature-discussion (RFC)

* Swiper Version: v5
* Live Link or JSFiddle/Codepen or website with isssue: https://codepen.io/italosimpli/pen/zYrLmLV

### What you did

Using **Auto Slides Per View / Carousel Mode** width different width for the `swiper-slide-active`

Relevant **JS**:
```javascript
    slidesPerView: 'auto',
    loop: true,
    loopedSlides: 20,
```

Relevant **CSS**:

```css
.swiper-slide {
    width: 200px;
    min-width: 200px;
    transition: min-width 0.3s ease;
}

.swiper-slide-active {
    min-width: 400px;
}
```

### Expected Behavior

When resizing the window, swipershould keep the initial behavior.

### Actual Behavior

After resizing the widow, all the slides right **after** the active one, has the `offset` values recalculated, and then the behavior of the swiper is broken.

I know the cause, but could not find a way to work around.

## Helpfull information

This behavior is result of the different widths. If we compare the initial `myswiper.slidesGrid` values and the values after resizing, we can see that the values are different:

```javascript
Initial Values for myswiper.slidesGrid = [0, 210, 420, 630, 840, 1050, 1260, 1470, 1680, 1890, 2100, 2310];
New Values for myswiper.slidesGrid   = [0, 210, 420, 630, 840, 1250, 1460, 1670, 1880, 2090, 2300, 2510];
```

Notice that `1050` is now `1250`.

The new values are not wrong, they are actually the right values, but they break the swiper behaviour, and I would like to keep the values as the initial, or be able to set some `slideIntrinsicWidth` or something like `translateVariation` value.

Basically the perfect behaviour for this particular case would be ignore the `active` width when calculating the `slidesGrid` values and consequently the `translate` value.

**Comment by nolimits4web** [comment]
The demo link you shared doesn't load for me. Can you do another one?

**Comment by italosimpli** [comment]
Sorry about that. Looks like i am having some issues with Stackblitz.

Here is the link, let me know if this works: https://codepen.io/italosimpli/pen/zYrLmLV (I also updated the link on the issue description)

**Comment by italosimpli** [comment]
Hi @nolimits4web, just a reminder about this issue.

Please let me know if you can see my demo now and if you or any other user have some fix or suggestion about how to get this working as expected.

**Comment by italosimpli** [comment]
I figured out a simple workaround for my case, but it only works if you are using `px` for the `swiper-slides` width, if the width is in `%` or `auto` this workaround will not work.

I just added `updateOnWindowResize: false` to the configurations, it will prevent the `slidesGird` to be updated keeping the initial values. This only works because `slidesGrid` is calculated before the classes are applied, so the width for the active slide (`swiper-slide-active`) is not considered (what is good for my use case).

Here are the main pieces to reproduce it and the Pen for it:

**Pen:** https://codepen.io/italosimpli1/pen/ZEpZmwv

**Main code pieces: **

JS
```javascript
 var swiper = new Swiper('.swiper-container', {
      
      updateOnWindowResize: false, // this prevents slideGrid to update, comment this line to see the initial issue
      
      slidesPerView: 'auto',
      loop: true,
      loopedSlides: 20,
      
      spaceBetween: 10,
      pagination: {
        el: '.swiper-pagination',
        clickable: true,
      },
    });
```

CSS:
```css
.swiper-slide {
    width: 200px;
    min-width: 200px;
    transition: min-width 0.3s ease;
}

.swiper-slide-active {
    min-width: 400px;
}
```

> **NOTE**: Although the slider seems to be working fine, it still have a little jump when it loops (see the following video), I am still not sure how to fix it, any suggestion is appreciated.

https://user-images.githubusercontent.com/59578655/105416341-d5aa8b80-5bee-11eb-91c6-2a80b5ed5bc3.mov

**Comment by milenig** [comment]
+1
